polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -870,6 +870,26 @@ ValueCategory MLIRScanner::VisitConstructCommon(clang::CXXConstructExpr *Cons,
 
 ValueCategory
 MLIRScanner::VisitArraySubscriptExpr(clang::ArraySubscriptExpr *Expr) {
+  LLVM_DEBUG({
+    llvm::dbgs() << "VisitArraySubscriptExpr: ";
+    Expr->dump();
+    llvm::dbgs() << "\n";
+  });
+
+  assert(!Expr->getBase()->getType()->isVLSTBuiltinType() &&
+         "Not supported yet");
+
+  if (Expr->getBase()->getType()->isVectorType()) {
```

Comment:
Would it make sense to have a set of private member functions to deal with the case where `Expr` has vector type. This function could delegate code generation to those functions when the expr is a vector.

Reply:
Done